app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -4,7 +4,13 @@ class OmniauthCallbacksController < ApplicationController
   def callback
     auth = request.env["omniauth.auth"]
 
-    session[:user_info] = auth.extra.raw_info
+    DfeIdentityCallbackForm.new(
```

Comment:
Suggestion: It's odd to use a form object for callback handling - perhaps we should call this `SignInWithDfeIdentityForm` to reflect that it's part of the handling of the preceding form where the user selects where or not to sign in. That would make a bit more sense to me.

However, I'm not sure there's any need to set `teacher_id_user_info` in the claim unless the user has confirmed their details are correct on the `teacher-detail` page. I think in the case they select 'No' we would not need to retain it. So potentially we could just keep this in the session until we decide to store it on the claim (on the `teacher-detail` form).

Reply:
`SignInWithDfeIdentityForm` - Yeah I think that's a better name 👍 

> However, I'm not sure there's any need to set teacher_id_user_info in the claim unless the user has confirmed their details are correct on the teacher-detail page

Yeah it is a bit odd storing this before confirming it's correct, though that was the [previous behaviour](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/4a9320e66b55dd03f91766061aadaef5eb031e5f/app/controllers/claims_controller.rb#L39) before this form was introduced and it seems like we don't clear the `teacher_id_user_info` when [details are incorrect](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/4a9320e66b55dd03f91766061aadaef5eb031e5f/app/models/dfe_identity/claim_user_details_reset.rb#L14). It looks like we always stored this info on the claim in a json column but only transferred it over to the claim's attributes proper (eg `first_name`, `surname`) if the teacher confirmed their details. Not sure if that's something we want to keep doing?

Reply:
🤔 An option to avoid storing these details in the claim until they're approved* is to redirect to the "teacher-details" page with the values from oauth as parameters

```ruby
    redirect_to claim_path(
      journey: current_journey_routing_name,
      slug: "teacher-detail",
      claim: {
        details_check: nil,
        logged_in_with_tid: true,
        teacher_id_user_info: auth.extra.raw_info.to_h.with_indifferent_access
      }
    )
```
then the `TeacherDetailForm` can read the values from the params, present them to the Teacher for them to confirm and store them on the claim if they do / loose them if they don't. This might be the most "railsy" way to do this as we're not changing any data in a GET request, though I'm not sure about GDPR with putting these details in the url.


*though I think that should be a separate PR as it seems `teacher_id_user_info` is used elsewhere

Reply:
Clever idea, but I'm a bit worried about logging as it would be a GET request, and we won't control the list of parameters that could slip through filtering.

Reply:
> Yeah it is a bit odd storing this before confirming it's correct, though that was the [previous behaviour](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/4a9320e66b55dd03f91766061aadaef5eb031e5f/app/controllers/claims_controller.rb#L39) before this form was introduced and it seems like we don't clear the `teacher_id_user_info` when [details are incorrect](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/4a9320e66b55dd03f91766061aadaef5eb031e5f/app/models/dfe_identity/claim_user_details_reset.rb#L14). It looks like we always stored the teacher_id_user_info on the claim in a json column but only transferred it over to the claim's attributes proper (eg `first_name`, `surname`) if the teacher confirmed their details. Not sure if that's something we want to keep doing? Didn't want to change any behaviour in this pr!

I don't think there is any need to store it in that scenario, and if it makes it easier and makes more sense then we should change it.

Reply:
> I don't think there is any need to store it in that scenario, and if it makes it easier and makes more sense then we should change it.

Do you want this doing as part of this PR or do you want me to do it as a separate ticket? My thoughts on doing this as a separate ticket is that we use `current_claim.teacher_id_user_info["email"]` in [a few places](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/cbef1edd799cc32bc79b409e69ef5acaec39c53a/app/controllers/claims_controller.rb#L48) but this doesn't depend on whether or not the details are confirmed as correct or not, as `teacher_id_user_info` [isn't cleared if the teacher says the details don't match](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/cbef1edd799cc32bc79b409e69ef5acaec39c53a/app/models/dfe_identity/claim_user_details_reset.rb#L2) so I'm unsure on the size of the changes required to clear this. Happy to do this either as part of this PR or a separate ticket, which ever you prefer 👍